presto-orc/src/main/java/com/facebook/presto/orc/metadata/OrcMetadataWriter.java

```diff
@@ -283,6 +283,7 @@ public int writeStripeFooter(SliceOutput output, StripeFooter footer)
 
     private static OrcProto.Stream toStream(Stream stream)
     {
+        checkArgument(stream.getSequence() == ColumnEncoding.DEFAULT_SEQUENCE_ID, "Writing streams with non-zero sequence IDs is not supported in ORC " + stream);
```

Comment:
+ stream will always call stream.toString() which will create garbage.  Remove the + stream or use format specifier.

Reply:
Stream has custom toString already implemented.

Reply:
nit: use format instead of concat

Reply:
It will print the right details (some readable string, instead of some memory address), that is not a concern.  Calling toString will create a temporary string and lot of string concatenation. We want to avoid that. 

two ways to avoid it are
1. Do not include the stream in the error message. less perfereable, as we will not see what stream failed.
2. call checkArgument(condition, "some message {}", object). Here object.toString() will be only called condition is false and avoiding the garbage on the hot path.

Reply:
Done. Also fixed the checkArgument in toColumnEncoding() to use format instead of concat